lib/_tls_wrap.js

```diff
@@ -423,6 +423,10 @@ function onerror(err) {
   if (!owner._secureEstablished) {
     // When handshake fails control is not yet released,
     // so self._tlsError will return null instead of actual error
+
+    // Set closing the socket after emitting an event since the socket needs to
+    // be accessible when the `tlsClientError` event is emmited.
+    owner._closeAfterHandlingError = true;
```

Comment:
Does this work with TLS connections over `Duplex` streams that are not `net.Socket`s?

Reply:
For that parts, I entirely relied on test cases. Could you walk me through any example to verify it?